src/searchBar.js

```diff
@@ -0,0 +1,20 @@
+import { updateUI_categoryPage } from './displayCategory';
+import { displayOne } from './displayOne';
+
+//Search tab functionality
+export function searchCategory(arr) {
+  const searchBar = document.querySelector("#searchBar");
+
+  updateUI_categoryPage(arr);
+  console.log('44444');
+
+  searchBar.addEventListener("input", (e) => {
+    e.preventDefault();
+    document.querySelector("#categories").innerHTML = "";
+    arr.filter((el) => {
```

Comment:
Again don't use filter as for/foreach loop, because filter has retrived result, for looping use loop for/foreach/forin, the same mistake like you used map above.